core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -7346,6 +7484,54 @@ private void handleRoleAlreadyExistExceptionWithID(String roleName, String[] use
         return finalValues;
     }
 
+    /**
+     * Handles the processing of any special user store attribute values after retrieval.
+     *
+     * @param userName       Username of the user.
+     * @param userAttributes un-processed map (user store attribute name -> attribute value) of user store
+     * @param profileName    Profile name of the user.
+     */
+    protected void processAttributesAfterRetrieval(String userName, Map<String, String> userAttributes,
+                                                   String profileName) {
+        // Not implemented for AbstractUserStoreManager, may have implementations at subclasses.
+    }
+
+    /**
+     * Handles the processing of any special user store attribute values before update.
+     *
+     * @param userName       Username of the user.
+     * @param userAttributes un-processed map (user store attribute name -> attribute value) of user store
```

Comment:
start with a capital and end with ".", refactor in the following methods as well